rust/lancedb/src/table.rs

```diff
@@ -1457,6 +1453,12 @@ impl NativeTable {
             }?
         };
 
+        let num_partitions = if let Some(n) = index.num_partitions {
+            n
+        } else {
+            suggested_num_partitions_for_hnsw(self.count_rows(None).await?, num_sub_vectors)
```

Comment:
I think `dims` was meant to be original dimensions, not the compressed subvectors. Could you get that? It's the `n` inside of the `arrow_schema::DataType::FixedSizeList`